### PR TITLE
Close #65

### LINKED ISSUE #65: General Issue: CRITxx utility
### Background Information

To create a new utility for CRITERIA/ANALYSIS flags.
The aim is to build up a function for deriving both character/numeric and label variables.

### Definition of Done

_No response_

### PR DESCRIPTION
**Pull request by federicobaratin**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page.
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging.
- [x] Address all merge conflicts and resolve appropriately.
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/derive_vars_crit.R [hunk 1]
```diff
@@ -0,0 +1,89 @@
+#' derive_vars_crit
```
Comment (bms63) [inline R/derive_vars_crit.R:1]: You need a Title for this function

Comment (bms63) [reply R/derive_vars_crit.R:1]: https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#function-header-documentation

**Review by bms63 — COMMENTED** [review]
@federicobaratin after you create your function run `devtools:document()` to make the corresponding `.Rd` file

**Comment by bms63** [comment]
Run `lintr::lint_package()` in R and it will give you advice on how to fix the linter check

**Comment by federicobaratin** [comment]
Hi Ben!
Added Title.
Created Rd file.
Removed warnigns for lintr package.
Thanks!

**Comment by bms63** [comment]
Hi @pharmaverse/admiralvaccines  You can use this function `spelling::update_wordlist()` to help deal with the failing Spelling GitHub Action.  This will add new words in your PR to `inst/WORDLIST`.  Please double check that your newly added words are not mis-spellings!!

File: R/derive_vars_crit.R [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+#' @author Federico Baratin
+#'
+#' @examples
+#'
+#' is2 <- derive_vars_crit(
+#'   dataset = is,
+#'   new_var = "CRIT1",
+#'   label_var = "Titer >= ISLLOQ",
+#'   condition = !is.na(AVAL) & !is.na(ISLLOQ),
+#'   criterion = AVAL >= ISLLOQ
```
Comment (bms63) [inline R/derive_vars_crit.R:50]: where is it getting the is data from - the package isn't smart enough to find data?  You have to supply it yourself - you can make a dummy example using `tibble::tribble()` or call `data("is")` 

Check out `derive_locf_records()` for example.  

https://github.com/pharmaverse/admiral/blob/main/R/derive_locf_records.R

image

Comment (federicobaratin) [reply R/derive_vars_crit.R]: Yeah, you're right.
Updated code with tribble example.
Also added family, keywords and export.

**Comment by bms63** [comment]
This one is good - merging in

**Pull request merged**